cloudinit/sources/DataSourceAzure.py

```diff
@@ -1945,6 +1948,28 @@ def load_azure_ds_dir(source_dir):
     return (md, ud, cfg, {"ovf-env.xml": contents})
 
 
+def configure_ephemeral_and_hotplug_device(netconfig: dict) -> dict:
```

Comment:
We should probably rename this function to something like `add_dhcp_config_for_primary_and_secondary_nics_to_network_config`. The doc string should be updated accordingly.